src/main/kotlin/racingcar/Car.kt

```diff
@@ -1,16 +1,31 @@
 package racingcar
 
 data class Car(
-    var location: Int = 0,
+    var location: Int = CAR_DEFAULT_LOCATION,
+    val name: String,
     private val engine: Engine = RandomPowerEngine(),
-    private val wheel: Wheel = Wheel()
 ) {
-    fun move() {
+    init {
+        require(this.name.length <= CAR_MAX_NAME_LENGTH) { "자동차 이름은 최대 ${CAR_MAX_NAME_LENGTH}글자 입니다" }
+    }
+
+    fun moveOrStop() {
         val power = engine.powerUp()
-        location += wheel.nextLocation(power)
+        location += if (isMovablePower(power)) 1 else 0
```

Comment:
개인적으로 아래와 같이 표현하는게 조금 더 가독성이 좋은 것 같아요. 🤔
```suggestion
        if (isMovablePower(power)) {
            this.location++
        }
```